testonly/integration/clientserver.go

```diff
@@ -115,62 +115,95 @@ func getTestDB(testID string) (*sql.DB, error) {
 	return dbTest, nil
 }
 
-// NewLogEnv creates a fresh DB, log server, and client.
+// NewLogEnv creates a fresh DB, log server, and client. The numSequencers parameter
+// indicates how many sequencers to run in parallel; if numSequencers is zero a
+// manually-controlled test sequencer is used.
 // testID should be unique to each unittest package so as to allow parallel tests.
-func NewLogEnv(testID string) (*LogEnv, error) {
+func NewLogEnv(ctx context.Context, numSequencers int, testID string) (*LogEnv, error) {
 	db, err := getTestDB(testID)
 	if err != nil {
 		return nil, err
 	}
 
 	timesource := &util.SystemTimeSource{}
-	registry, err := builtin.NewDefaultExtensionRegistry()
+	registry, err := builtin.NewExtensionRegistry(db)
 	if err != nil {
 		return nil, err
 	}
 
-	// Start Log Server.
+	// Create Log Server.
 	grpcServer := grpc.NewServer()
 	logServer := server.NewTrillianLogRPCServer(registry, timesource)
 	trillian.RegisterTrillianLogServer(grpcServer, logServer)
 
-	// Start Sequencer.
+	// Create Sequencer.
 	keyManager, err := crypto.LoadPasswordProtectedPrivateKey(privateKeyFile, privateKeyPassword)
 	if err != nil {
 		return nil, err
 	}
-	sequencerManager := server.NewSequencerManager(keyManager, registry,
-		sequencerWindow)
-	sequencerTask := server.NewLogOperationManagerForTest(ctx, registry,
-		batchSize, sleep, timeSource, sequencerManager)
+	sequencerManager := server.NewSequencerManager(keyManager, registry, sequencerWindow)
+	var wg sync.WaitGroup
+	var sequencerTask *server.LogOperationManager
+	var cancel context.CancelFunc
+	if numSequencers == 0 {
+		// Test sequencer that needs manual triggering (with env.Sequencer.OperationLoop()).
+		sequencerTask = server.NewLogOperationManagerForTest(ctx, registry,
+			batchSize, sleepBetweenRuns, timeSource, sequencerManager)
+	} else {
+		// Start a live sequencer in a goroutine.
+		var ctx2 context.Context
+		ctx2, cancel = context.WithCancel(ctx)
+		sequencerTask = server.NewLogOperationManager(ctx2, registry,
+			batchSize, numSequencers, sleepBetweenRuns, timesource, sequencerManager)
+		wg.Add(1)
+		go func(wg *sync.WaitGroup, om *server.LogOperationManager) {
```

Comment:
Rather than adding the sequencer to a wait group, is there a way we could create a `GracefulStop` to the log operation manager as well?  

It seems to me that the management of what sequencer runs are active etc, and how to do a proper shutdown should be internal to the log operation manager.